app/models/manageiq/providers/kubevirt/inventory/parser.rb

```diff
@@ -82,6 +83,18 @@ def process_node(object)
     os_object.product_type = node_info.operatingSystem
     os_object.version = node_info.kernelVersion
 
+    hw_object = host_hw_collection.find_or_build(host_object)
+
+    memory = object.status&.capacity&.memory
+    memory &&= begin
+      memory.iec_60027_2_to_i
+    rescue
+      memory.decimal_si_to_f
+    end
+
+    hw_object.memory_mb = memory / 1.megabyte if memory
```

Comment:
I can't tell, but is it possible for this to end up with integer division if memory is an integer?  Should we force float division here with something like?

```suggestion
    hw_object.memory_mb = memory / 1.megabyte.to_f if memory
```

Reply:
Ok then I have the opposite concern in that memory could be a float from `decimal_si_to_f`

Reply:
Pretty sure AR assignment to a bigint column would floor that for us, interestingly this is what we do in the VMware parser `(memory_size.to_f / 1.megabyte).round`, also this specific logic above was ~blatantly ripped off~ "borrowed" from the kubernetes parser.  If you want we could `(memory.to_f / 1.megabyte).floor` here and in kubernetes but I'm not sure if you can have a fractional `Mi` value in k8s (easy to test....)

Reply:
yeah I think we should be explicit, so the .to_f .floor seems appropriate